elk/utils/data_utils.py

```diff
@@ -131,14 +132,14 @@ def undersample(
     return dataset
 
 
-def float32_to_int16(x: torch.Tensor) -> torch.Tensor:
-    """Converts float32 to float16, then reinterprets as int16."""
-    return x.type(torch.float16).view(torch.int16)
-
-
-def int16_to_float32(x: torch.Tensor) -> torch.Tensor:
-    """Converts int16 to float16, then reinterprets as float32."""
-    return x.view(torch.float16).type(torch.float32)
+def get_layers(ds: DatasetDict) -> List[int]:
+    """Get a list of indices of hidden layers given a `DatasetDict`."""
+    layers = [
+        int(feat[len("hidden_") :])
+        for feat in ds["train"].features
```

Comment:
ds might not contain a "train" split, so we should just take ds.values().pop().features or the equivalent.

Reply:
don't you always still get at least train if there is no split?

Reply:
I'm thinking about the refactor I want to do to support evaluations on a single split, and in the case of E.g. truthful QA there is only a validation split.